### PR TITLE
refactor: Implement osName, remove os-name package

### PR DESCRIPTION
**Pull request by chamsou123**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
`nest info` logs `Unknown` on newer versions of os versions which are not added yet to os-name package, while os-name is not updated frequently we can't depend on it.

Issue Number: #2400

## What is the new behavior?
`nest info` now logs `MacOs 23.0.0` instead of 'MacOs Sonoma' for example.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

### CODE REVIEW

File: actions/info.action.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export class InfoAction extends AbstractAction {
 
   private async displaySystemInformation(): Promise<void> {
     console.info(chalk.green('[System Information]'));
-    console.info('OS Version     :', chalk.blue(osName(platform(), release())));
+    console.info('OS Version     :', chalk.blue(osName(platform(), release()), release()));
```
Comment (Tony133) [inline actions/info.action.ts:42]:
```suggestion
    console.info('OS Version     :', chalk.blue(osName(platform(), release())));
```

**Review by Tony133 — CHANGES_REQUESTED** [review]

File: actions/info.action.ts (hunk 1 above)
Comment (chamsou123) [reply actions/info.action.ts:42]: @Tony133 this will log `OS Version : MacOs` instead of `OS Version : MacOs 23.0.0`

Comment (Tony133) [reply actions/info.action.ts:42]: Ah okay, it looks like a repetition of the method

**Review by micalevisk — APPROVED** [review]

**Review by Tony133 — APPROVED** [review]

**Review by rubiin — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
lgtm

**Pull request merged**